docs/machine-learning/how-to-guides/train-machine-learning-model-cross-validation-ml-net.md

```diff
@@ -0,0 +1,116 @@
+
+## The data and data model
+
+Given data from a file that has the following format:
+
+```text
+Size (Sq. ft.), HistoricalPrice1 ($), HistoricalPrice2 ($), HistoricalPrice3 ($), Current Price ($)
+620.00, 148330.32, 140913.81, 136686.39, 146105.37
+550.00, 557033.46, 529181.78, 513306.33, 548677.95
+1127.00, 479320.99, 455354.94, 441694.30, 472131.18
+1120.00, 47504.98, 45129.73, 43775.84, 46792.41
+```
+
+The data can be modeled by a class like `HousingData`:
+
+```csharp
+public class HousingData
+{
+    [LoadColumn(0)]
+    public float Size { get; set; }
+ 
+    [LoadColumn(1, 3)]
+    [VectorType(3)]
+    public float[] HistoricalPrices { get; set; }
+
+    [LoadColumn(4)]
+    [ColumnName("Label")]
+    public float CurrentPrice { get; set; }
+}
+```
+
+Load the data in into an [`IDataView`](xref:Microsoft.ML.IDataView).
+
+## Prepare the data
+
+Pre-process the data before using it to build the machine learning model. In this sample, the `Size` and `HistoricalPrices` columns are combined into a single feature vector,  which is output to a new column called `Features` using the [`Concatenate`](xref:Microsoft.ML.TransformExtensionsCatalog.Concatenate*) method. Then, [`NormalizeMinMax`](xref:Microsoft.ML.NormalizationCatalog.NormalizeMinMax*) is applied to the `Features` column to get `Size` and `HistoricalPrices` in the same range between 0-1.
+
+```csharp
+// Define data prep estimator
+IEstimator<ITransformer> dataPrepEstimator = 
+    mlContext.Transforms.Concatenate("Features", new string[] { "Size", "HistoricalPrices" })
+        .Append(mlContext.Transforms.NormalizeMinMax("Features"));
+
+// Create data prep transformer
+ITransformer dataPrepTransformer = dataPrepEstimator.Fit(data);
+
+// Transform data
+IDataView transformedData = dataPrepTransformer.Transform(data);
+```
+
+## Train model with cross validation
+
+Once the data has been pre-processed, it's time to train the model. First, select the algorithm that most closely aligns with the machine learning task to be performed. Because the predicted value is a numerically continuous value, the task is regression. One of the regression algorithms implemented by ML.NET is the [`StochasticDualCoordinateAscentCoordinator`](xref:Microsoft.ML.Trainers.SdcaRegressionTrainer) algorithm. To train the model with cross-validation use the [`CrossValidate`](xref:Microsoft.ML.RegressionCatalog.CrossValidate*) method.
+
+> [!NOTE]
+> CrossValidate is also available for clustering, binary classification and multiclass classification algorithms.
+
+```csharp
+// Define StochasticDualCoordinateAscent algorithm estimator
+IEstimator<ITransformer> sdcaEstimator = mlContext.Regression.Trainers.Sdca();
+
+// Apply 5-fold cross validation
+var cvResults = mlContext.Regression.CrossValidate(transformedData, sdcaEstimator, numberOfFolds: 5);
```

Comment:
>numberOfFolds: 5); [](start = 83, length = 18)

might be good to also set the stratification column, but it wouldn't apply to this example..

Reply:
When / How would you use the stratification column (if possible)?